File: 0009-Palindrome Number/0009-Palindrome Number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0)
        {
            return false;
        }
        
        int power = 1;
        while (x / power >= 10)
        {
            power *= 10;
        }
        
        while (power > 1)
        {
            int num1 = x % 10;
            int num2 = x / power;
            if (num1 != num2)
            {
                return false;
            }
            
            x = x % power / 10;
            power /= 100;
        }
        
        return true;
    }
};

File: 0001-0100/0009-Palindrome Number/0009-Palindrome Number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0 or (x != 0 and x%10 == 0):
            return False
        y = 0
        while x > y:
            y = y*10 + x%10
            x //= 10
        return x == y or y//10 == x
